.github/workflows/ci_sanitizer.yml

```diff
@@ -0,0 +1,111 @@
+# SPDX-FileCopyrightText: 2006-2024, Knut Reinert & Freie Universität Berlin
+# SPDX-FileCopyrightText: 2016-2024, Knut Reinert & MPI für molekulare Genetik
+# SPDX-License-Identifier: CC0-1.0
+
+name: Sanitizer
+
+on:
+  schedule:
+    - cron: "0 6 * * SAT"
+  workflow_dispatch:
+
+concurrency:
+  group: sanitizer-actions
+  cancel-in-progress: true
+
+env:
+  SEQAN3_NO_VERSION_CHECK: 1
+  TZ: Europe/Berlin
+  TSAN_OPTIONS: ignore_noninstrumented_modules=1
+  UBSAN_OPTIONS: print_stacktrace=1
+
+defaults:
+  run:
+    shell: bash -Eeuxo pipefail {0}
+
+jobs:
+  build:
+    name: ${{ matrix.name }} ${{ matrix.build_type }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    if: github.repository_owner == 'seqan' || github.event_name == 'workflow_dispatch'
+    env:
+      ASAN_OPTIONS: strict_string_checks=1:detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1:detect_leaks=${{ contains(matrix.os, 'macos') && '0' || '1' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ASan, TSan, UBSan]
+        os: [ubuntu-latest, macos-14]
+        build_type: [Release, RelWithDebInfo, Debug]
+        exclude:
+          # macOS llvm packages do not contain libarcher, which is required for TSan to handle OpenMP.
+          # TSan runs on ubuntu with clang. Packages there contain libarcher.
+          - name: "TSan"
+            os: macos-14
+
+        include:
+          - os: macos-14
+            compiler: clang-19
+          - os: ubuntu-latest
+            compiler: gcc-14
+            image: ghcr.io/seqan/gcc-14
+
+          - name: "TSan"
+            os: ubuntu-latest
+            compiler: clang-19
+            image: ghcr.io/seqan/clang-19
+            cxx_flags: "-fsanitize=thread"
+            ctest_excludes: "-E async_input_buffer_snippet"
+
+          - name: "ASan"
+            os: ubuntu-latest
+            cxx_flags: "-fsanitize=address -Wno-maybe-uninitialized"
+          - name: "ASan"
+            os: macos-14
+            cxx_flags: "-fsanitize=address"
+
+          - name: "UBSan"
+            os: ubuntu-latest
+            cxx_flags: "-fsanitize=undefined,float-divide-by-zero -Wno-maybe-uninitialized -Wno-stringop-overflow"
+          - name: "UBSan"
+            os: macos-14
+            cxx_flags: "-fsanitize=undefined,float-divide-by-zero,local-bounds,nullability"
+            ctest_excludes: "-E tmp_directory_snippet_cmp_output"
+
+    container:
+      image: ${{ matrix.image || '' }}
```

Comment:
If `matrix.image` is defined for a job, it will be used.
If not, it will be empty; which means that no container is used.

There is no nice syntax for such things yet, but 
```
container:
      image: ${{ matrix.image || '' }}
```
means
```
if (image is defined for job):
    run job in container using the image
else:
    use no container
```

Reply:
Internally, it's parsed like this, I think:

`${{ matrix.image || '' }}` -> `matrix.image ? matrix.image : ''`

Any (non-empty?) string is a truthy value.
Undefined (and empty string?) values are false.

If I recall correctly, `image: matrix.image` wouldn't work because there is a difference between undefined values and empty strings. Or something like that :)

Reply:
Anyway, I added a comment locally:

```
    container:
      # If an image is defined for a matrix entry, use it.
      # Otherwise, use the "empty"/'' image which means do not use a container at all.
      image: ${{ matrix.image || '' }}
      volumes:
        - /home/runner:/home/runner
```

I'm just waiting with force-pushing in case there are other review comments.